app/black_box/black_box.rb

```diff
@@ -32,14 +32,17 @@ def comment_quality_score(comment)
       (rep_points + descendants_points + bonus_points - spaminess_rating).to_i
     end
 
-    def calculate_spaminess(story, function_caller = FunctionCaller)
-      # accepts comment or article as story
-      return 100 unless story.user
-      return 0 if ENV["AWS_SDK_KEY"].blank? # Skip this if we don't have a private spam score for now
-      return 0 if ENV["AWS_SDK_KEY"] == "foobarbaz" # Also skip if placeholder
+    def calculate_spaminess(story)
+      user = story.user
+      return 100 unless user
+      return 100 if user.trusted
+      return 100 if user.badges_count.positive?
 
-      function_caller.call("blackbox-production-spamScore",
-                           { story: story, user: story.user }.to_json).to_i
+      base_spaminess = 0
+      base_spaminess += 25 if
+        user.registred_at < ((user.github_created_at || user.twitter_created_at || 3.days.ago) + 2.days) &&
```

Comment:
there's a typo here in the name of the attribute `registered_at`

I wonder if you could use `user.registered_at.between?(min, max)` for readability

Reply:
This is actually asking for the relationship between registered at and the recency of the social log in— which probably speaks to the readability issues in the first place.

Reply:
Also, I think we should change this as `github_created_at` and `twitter_created_at` might not exist for all Forems.

You should be able to do something like:

```ruby
enabled_providers = Authentication::Providers.enabled_for_user(user)
enabled_providers.map(&:user_created_at_field).map { |f| user.public_send(f) }
```

when we merge https://github.com/forem/forem/pull/9961